.github/workflows/create-builds.yml

```diff
@@ -0,0 +1,109 @@
+        required: true
+        type: choice
+        # We only allow internal and release candidate builds when doing manual dispatches.
+        # Production builds should only be triggered by the appropriate calling workflows.
+        options: ['internal', 'release-candidate']
+
+jobs:
+  validate_inputs:
+    name: Validate inputs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate app_type input
+        if: ${{ !(inputs.app_type == 'internal' || inputs.app_type == 'release-candidate' || inputs.app_type == 'production') }}
+        run: |
+          echo "app_type input must be one of the following: internal, release-candidate, production"
+          exit 1
+
+  builds:
+    name: Create builds
+    needs: validate_inputs
+    strategy:
+      matrix:
+        os:
+          [
+            { name: 'linux', image: 'ubuntu-latest' },
+            { name: 'macos', image: 'macos-latest' },
+            { name: 'macos-intel', image: 'macos-13' },
+            { name: 'windows', image: 'windows-latest' },
+          ]
+    runs-on: ${{ matrix.os.image }}
+    steps:
+      - name: Create GitHub App Token
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.RELEASE_BOT_APP_ID }}
+          private-key: ${{ secrets.RELEASE_BOT_PRIVATE_KEY }}
+
+      - name: Checkout repository
+        env:
+          # When called from another workflow, the input value is used.
+          # When called from a manual dispatch, uses the value corresponding to the branch/tag selected in the GitHub UI.
+          BUILD_SHA: ${{ inputs.build_sha || github.sha }}
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BUILD_SHA }}
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build translations
+        run: npm run intl:translations
+
+      - name: Create distributables
+        env:
+          APP_TYPE: ${{ inputs.app_type }}
+          ONLINE_STYLE_URL: ${{ secrets.ONLINE_STYLE_URL }}
+          VITE_FEATURE_TEST_DATA_UI: ${{ env.APP_TYPE == 'production' && 'false' || 'true' }}
+        run: npm run forge:make
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.os.name }}-${{ inputs.build_sha }}
+          compression-level: 0
+          overwrite: true
+          path: |
+            out/make/**/RELEASES
+            out/make/**/*.AppImage
+            out/make/**/*.exe
+            out/make/**/*.dmg
+            out/make/**/*.nupkg
+            out/make/**/*.rpm
+            out/make/**/*.zip
+
+      - name: Publish application
+        if: ${{ inputs.APP_TYPE == 'production' }}
+        env:
+          GITHUB_TOKEN: ${{ steps.app-token.outputs.token }}
+        run: npm run forge:publish
```

Comment:
kind of feels like this should be in release.yml, because it depends on the release creation in that step (the contract to not run this workflow outside of release.yml for app_type production feels very loose). However that can be a follow up - might be tricky to pass the correct artifacts etc. You could make the create build an action like parse-version so it can be run in a single step. Anyway, fine like this for now.